lib/util/git/parent-sha-cache.ts

```diff
@@ -13,3 +14,22 @@ export function getCachedBranchParentShaResult(
 
   return null;
 }
+
+export function setCachedBranchParentShaResult(
+  branchName: string,
+  parentSha: string
+): void {
+  const cache = getCache();
+  cache.branches ??= [];
+
+  let branch = cache.branches.find(
+    (branch) => branch.branchName === branchName
+  );
+  if (!branch) {
+    branch = {
+      branchName: branchName,
+    } as BranchCache;
+    cache.branches.push(branch);
+  }
+  branch.parentSha = parentSha;
```

Comment:
If the parentSha changes, shouldn't the cached isConflicted and isStale results be removed or recalculated?